src/Shared/Debugging/DebugUtils.cs

```diff
@@ -27,25 +27,27 @@ static DebugUtils()
         }
 
         // DebugUtils are initialized early on by the test runner - during preparing data for DataMemeberAttribute of some test,
-        //  for that reason it is not easily possible to inject the DebugPath in tests via env var (unless we want to run expensive exec style test).
+        // for that reason it is not easily possible to inject the DebugPath in tests via env var (unless we want to run expensive exec style test).
         internal static void SetDebugPath()
         {
             string environmentDebugPath = FileUtilities.TrimAndStripAnyQuotes(Environment.GetEnvironmentVariable("MSBUILDDEBUGPATH"));
             string debugDirectory = environmentDebugPath;
-
+            bool needsSubfolder = !string.IsNullOrWhiteSpace(environmentDebugPath) &&
+                          !debugDirectory.EndsWith(".MSBuild_Logs", StringComparison.OrdinalIgnoreCase);
             if (Traits.Instance.DebugEngine)
             {
-                if (!string.IsNullOrWhiteSpace(debugDirectory) && FileUtilities.CanWriteToDirectory(debugDirectory))
+                if (needsSubfolder && FileUtilities.CanWriteToDirectory(debugDirectory))
```

Comment:
This code does not seem correct: if the folder ends with ".MSBuild_Logs" already, `needsSubfolder` would be false and we would assign the fallback folder to be `debugDirectory`.
 
I also do not think that we need to skip adding the ".MSBuild_Logs" if the folder already ends with it. It might bring additional confusion about the behavior.